### ISSUE TITLE
How to log more metrics with wandb when using GRPO trainer and accelerate

### ISSUE THREAD

**Issue by andrewsiah** [huggingface/trl#2768]
### Reproduction

```python

def correctness_reward_func(prompts, completions, answer, **kwargs) -> list[float]:
    responses = [completion[0]["content"] for completion in completions]
    q = prompts[0][-1]["content"]
    extracted_responses = [extract_xml_answer(r) for r in responses]

    # Get current step from trainer's state
    current_step = trainer.state.global_step if hasattr(trainer, "state") else 0

    # Initialize logger if not already done
    global example_logger
    if not hasattr(correctness_reward_func, "example_logger"):
        example_logger = LocalExampleLogger()
        correctness_reward_func.example_logger = example_logger

    # Log each example
    for i in range(len(responses)):
        example_dict = {
            "step": current_step,
            "question": q,
            "true_answer": answer[i],
            "response": responses[i],
            "extracted_response": extracted_responses[i],
            "correct": extracted_responses[i] == answer[i],
            "generation_idx": i,  # Which generation attempt this was
        }
        example_logger.log_example(example_dict)

    # Calculate marker counts and correctness for all responses
    is_correct = [r == a for r, a in zip(extracted_responses, answer)]
    uncertainty_counts = [count_uncertainty_markers(r) for r in responses]
    internal_dialogue_counts = [count_internal_dialogue_markers(r) for r in responses]
    reflective_counts = [count_reflective_markers(r) for r in responses]

    # Separate counts for correct and incorrect responses
    correct_indices = [i for i, correct in enumerate(is_correct) if correct]
    incorrect_indices = [i for i, correct in enumerate(is_correct) if not correct]

    # Log metrics using trainer's accelerator
    if hasattr(trainer, "accelerator"):
        metrics = {
            "correctness/correct_count": len(correct_indices),
            "correctness/total_examples": len(responses),
            "correctness/accuracy": len(correct_indices) / len(responses),
            # Total markers across all responses
            "markers/total/uncertainty": sum(uncertainty_counts),
            "markers/total/internal_dialogue": sum(internal_dialogue_counts),
            "markers/total/reflective": sum(reflective_counts),
            # Markers in correct responses
            "markers/correct/uncertainty": sum(
                uncertainty_counts[i] for i in correct_indices
            )
            if correct_indices
            else 0,
            "markers/correct/internal_dialogue": sum(
                internal_dialogue_counts[i] for i in correct_indices
            )
            if correct_indices
            else 0,
            "markers/correct/reflective": sum(
                reflective_counts[i] for i in correct_indices
            )
            if correct_indices
            else 0,
            # Markers in incorrect responses
            "markers/incorrect/uncertainty": sum(
                uncertainty_counts[i] for i in incorrect_indices
            )
            if incorrect_indices
            else 0,
            "markers/incorrect/internal_dialogue": sum(
                internal_dialogue_counts[i] for i in incorrect_indices
            )
            if incorrect_indices
            else 0,
            "markers/incorrect/reflective": sum(
                reflective_counts[i] for i in incorrect_indices
            )
            if incorrect_indices
            else 0,
        }
        trainer.accelerator.log(metrics, step=current_step)

    return [2.0 if r == a else 0.0 for r, a in zip(extracted_responses, answer)]

.......

model_name = config["model"]["name"]
output_dir = config["training"]["output_dir"]
run_name = config["training"]["run_name"]

training_args = GRPOConfig(
    output_dir=output_dir,
    run_name=run_name,
    learning_rate=config["training"]["learning_rate"],
    adam_beta1=config["training"]["adam_beta1"],
    adam_beta2=config["training"]["adam_beta2"],
    weight_decay=config["training"]["weight_decay"],
    warmup_ratio=config["training"]["warmup_ratio"],
    lr_scheduler_type=config["training"]["lr_scheduler_type"],
    logging_steps=config["training"]["logging_steps"],
    bf16=config["training"]["bf16"],
    per_device_train_batch_size=config["training"]["per_device_train_batch_size"],
    gradient_accumulation_steps=config["training"]["gradient_accumulation_steps"],
    num_generations=config["training"]["num_generations"],
    max_prompt_length=config["training"]["max_prompt_length"],
    max_completion_length=config["training"]["max_completion_length"],
    num_train_epochs=config["training"]["num_train_epochs"],
    save_steps=config["training"]["save_steps"],
    max_grad_norm=config["training"]["max_grad_norm"],
    report_to=["wandb"]
    if (not torch.distributed.is_initialized() or torch.distributed.get_rank() == 0)
    else [],
    log_on_each_node=False,  # Only log on main node
    use_vllm=config["vllm"]["use_vllm"],
    vllm_gpu_memory_utilization=config["vllm"]["gpu_memory_utilization"],
    vllm_device=config["vllm"]["device"],
    gradient_checkpointing=config["training"]["gradient_checkpointing"],
)

model = AutoModelForCausalLM.from_pretrained(
    model_name,
    torch_dtype=getattr(torch, config["model"]["torch_dtype"]),
    attn_implementation=config["model"]["attn_implementation"],
    device_map=None,
    cache_dir=cache_dir,
).to("cuda")

tokenizer = AutoTokenizer.from_pretrained(model_name, cache_dir=cache_dir)
tokenizer.pad_token = tokenizer.eos_token

trainer = GRPOTrainer(
    model=model,
    processing_class=tokenizer,
    reward_funcs=[
        xmlcount_reward_func,
        soft_format_reward_func,
        strict_format_reward_func,
        int_reward_func,
        correctness_reward_func,
    ],
    args=training_args,
    train_dataset=dataset,
    eval_dataset=test_dataset,  # Add test dataset for evaluation
)

```

When I look at wandb, none of the extra metrics I added are logged.

How do we add more metrics to log with wandb, I tried declaring a new wandb.init(), but when using multi-GPU with accelerate, that then spins up N new wandb instances. 
So we know Accelerate (or Trainer) has an intrinsic wandb, how do we log with that?

### System Info

INFO 02-04 22:58:32 __init__.py:183] Automatically detected platform cuda.

Copy-paste the following information when reporting an issue:

- Platform: Linux-5.15.0-130-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version: 2.5.1
- CUDA device(s): NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB
- Transformers version: 4.48.2
- Accelerate version: 1.3.0
- Accelerate config: 
  - compute_environment: LOCAL_MACHINE
  - distributed_type: MULTI_GPU
  - mixed_precision: no
  - use_cpu: False
  - debug: False
  - num_processes: 2
  - machine_rank: 0
  - num_machines: 1
  - gpu_ids: all
  - rdzv_backend: static
  - same_network: True
  - main_training_function: main
  - enable_cpu_affinity: False
  - downcast_bf16: no
  - tpu_use_cluster: False
  - tpu_use_sudo: False
  - tpu_env: []
- Datasets version: 3.2.0
- HF Hub version: 0.28.1
- TRL version: 0.14.0
- bitsandbytes version: not installed
- DeepSpeed version: not installed
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.61.0
- PEFT version: not installed

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by qgallouedec** [comment]
Hey @andrewsiah, sorry for the very late reply, we're catching up on the GRPO backlog.

Your `trainer.accelerator.log(...)` calls never showed up because the HF `Trainer`'s wandb integration runs through `WandbCallback` (it calls `wandb.init()` on the main process and logs in `on_log`), not through Accelerate's trackers. The Trainer never calls `accelerator.init_trackers()`, so `accelerator.log()` has no registered tracker and silently no-ops. Calling `wandb.init()` yourself then races that callback, which is why you got N runs under multi-GPU.

Good news: current TRL supports exactly this: reward functions now receive a `log_metric(name, value)` callback (plus `trainer_state`, so the `global_step` hack is no longer needed) as kwargs; TRL averages the values across processes and the main process logs them alongside `kl`/`entropy`, so they land in wandb automatically with no manual `wandb.init()`. Just add `log_metric=None, **kwargs` to your `correctness_reward_func` and call e.g. `log_metric("correctness/accuracy", acc)`. See the documented example: https://github.com/huggingface/trl/blob/4f0b9b5ab17cd59dac39484d291f7b54e09e6868/docs/source/grpo_trainer.md#L596-L616.
